src/main/java/uk/gov/ons/ctp/response/collection/exercise/distribution/SampleUnitDistributor.java

```diff
@@ -131,15 +121,16 @@ public void distributeSampleUnits(CollectionExercise exercise) {
         return;
       }
 
-      for (ExerciseSampleUnitGroup sampleUnitGroup : sampleUnitGroups) {
-        try {
-          distributeSampleUnitGroup(exercise, sampleUnitGroup);
-        } catch (RestClientException ex) {
-          log.error(
-              "Failed to distribute sample unit group, sampleUnitGroupPK: {}",
-              sampleUnitGroup.getSampleUnitGroupPK());
-        }
-      }
+      sampleUnitGroups.forEach(
+          sampleUnitGroup -> {
+            try {
+              distributeSampleUnitGroup(exercise, sampleUnitGroup);
+            } catch (RestClientException ex) {
+              log.error(
```

Comment:
I know this is only a copy-paste job, but this will swallow an error and fail to roll back the transaction, which causes stability bugs